### ISSUE TITLE
[Build fail] WSL2 non-root filesystem, packages with npm dependencies.

### ISSUE THREAD

**Issue by brucejo75** [meteor/meteor#11320]
Getting build failure for meteor project not in the WSL2 root file system.  Shows up in packages that have `npm` dependencies.

## Environment
Meteor: **2.0**
Microsoft Windows 10 **Version 20H2 (OS Build 19042.804)**
WSL2 version reported from node `os.release()`: "**5.4.72-microsoft-standard-WSL2**"

## Build Fail
```
Error: EACCES: permission denied, rename
'/mnt/e/tests/meteor-package-test/packages/cow-say/.npm/package-new-1h0suxi.2hps' ->
'/mnt/e/tests/meteor-package-test/packages/cow-say/.npm/package'
```

<details>
<summary>Full Log Output</summary>

```log
brucejo:cow-say: updating npm dependencies -- cowsay...
/home/brucejo/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/meteor-promise/promise_server.js:218
      throw error;
      ^

Error: EACCES: permission denied, rename '/mnt/e/tests/meteor-package-test/packages/cow-say/.npm/package-new-1h0suxi.2hps' -> '/mnt/e/tests/meteor-package-test/packages/cow-say/.npm/package'
    at Object.renameSync (fs.js:741:3)
    at /home/brucejo/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/fs/tools/fs/files.ts:1638:23
    at Object.renameDirAlmostAtomically (/home/brucejo/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/fs/tools/fs/files.ts:1073:7)
    at completeNpmDirectory (/tools/isobuild/meteor-npm.js:811:9)
    at createFreshNpmDirectory (/tools/isobuild/meteor-npm.js:768:3)
    at Object.meteorNpm.updateDependencies (/tools/isobuild/meteor-npm.js:117:7)
    at Object.compile (/tools/isobuild/compiler.js:128:19)
    at /tools/isobuild/isopack-cache.js:354:30
    at Object.enterJob (/tools/utils/buildmessage.js:388:12)
    at IsopackCache._loadLocalPackage (/tools/isobuild/isopack-cache.js:309:18)
    at /tools/isobuild/isopack-cache.js:243:16
    at Function.time (/home/brucejo/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/tool-env/tools/tool-env/profile.ts:284:12)
    at /tools/isobuild/isopack-cache.js:242:17
    at Object.enterJob (/tools/utils/buildmessage.js:388:12)
    at IsopackCache._ensurePackageLoaded (/tools/isobuild/isopack-cache.js:233:20)
    at /tools/isobuild/isopack-cache.js:75:14
    at /tools/packaging/package-map.js:57:7
    at Function._.each._.forEach (/home/brucejo/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/underscore/underscore.js:191:9)
    at exports.PackageMap.eachPackage (/tools/packaging/package-map.js:49:7)
    at IsopackCache.buildLocalPackages (/tools/isobuild/isopack-cache.js:74:24)
    at /tools/project-context.js:890:25
    at Object.enterJob (/tools/utils/buildmessage.js:388:12)
    at ProjectContext._buildLocalPackages (/tools/project-context.js:889:18)
    at /tools/project-context.js:300:9
    at Object.enterJob (/tools/utils/buildmessage.js:388:12)
    at ProjectContext._completeStagesThrough (/tools/project-context.js:290:18)
    at /tools/project-context.js:282:12
    at Function.run (/home/brucejo/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/tool-env/tools/tool-env/profile.ts:289:14)
    at ProjectContext.prepareProjectForBuild (/tools/project-context.js:281:13)
    at /tools/runners/run-app.js:556:29
    at Object.capture (/tools/utils/buildmessage.js:283:5)
    at bundleApp (/tools/runners/run-app.js:555:31)
    at AppRunner._runOnce (/tools/runners/run-app.js:627:35)
    at AppRunner._fiber (/tools/runners/run-app.js:948:28)
    at /tools/runners/run-app.js:410:12 {
  errno: -13,
  syscall: 'rename',
  code: 'EACCES',
  path: '/mnt/e/tests/meteor-package-test/packages/cow-say/.npm/package-new-1h0suxi.2hps',
  dest: '/mnt/e/tests/meteor-package-test/packages/cow-say/.npm/package'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mpt@ start: `meteor run --exclude-archs 'web.browser.legacy, web.cordova'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mpt@ start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/brucejo/.npm/_logs/2021-02-15T22_58_38_478Z-debug.log
```
</details>

## Repro
See https://github.com/brucejo75/meteor-package-test.

## Problem code

The problem is in the function `isWindowsLikeFilesystem()`: https://github.com/meteor/meteor/blob/46e00a875726dff078c59eaf7fdc7b956cf0a996/tools/static-assets/server/mini-files.ts#L16-L18

Non-root filesystems in WSL are `WindowsLikeFilesystems` and require the [retry on renaming](https://github.com/meteor/meteor/blob/6e61b5bc9ca8a8170b5ef28832010d0c12c2f5f2/tools/fs/files.ts#L1706).

Apparently, in WSL2 Microsoft named the release differently than with WSL1.  Using lowercase `microsoft` in the release string that node produces.  This is easily remedied by changing the code in [`mini-files.ts:L17`](https://github.com/meteor/meteor/blob/46e00a875726dff078c59eaf7fdc7b956cf0a996/tools/static-assets/server/mini-files.ts#L17):

```diff
- return process.platform === "win32" || release().indexOf("Microsoft") >= 0;
---
+ return process.platform === "win32" || release().toLowerCase().indexOf("microsoft") >= 0;
```
## Workaround
After installing your local version of meteor change `mini-files.js` in `<your meteor install folder>/.meteor/packages/meteor-tool/.2.0.0.1xc5mk9.uwet++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/tools/static-assets/server/mini-files.js` as above.

**Comment by jeremydyoung** [comment]
Just ran into a problem with Meteor during a changed file update being unable to rename from a change in VS Code on WSL 2 with Ubuntu 20.04.

The change with toLowerCase() worked for me.

Thanks!

**Comment by filipenevola** [comment]
This is going to be included on 2.2
